http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/AppServiceInterp.scala

```diff
@@ -662,6 +676,77 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
+      )
+      landingZoneId = landingZone.landingZoneId
+
+      // Step 2: call LZ for LZ resources
+      lzResourcesByPurpose <- wsmDao.listLandingZoneResourcesByType(landingZoneId, userToken)
+
+      aksClusterName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                   "Microsoft.ContainerService/managedClusters",
+                                                   "SHARED_RESOURCE"
+      )
+      batchAccountName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                     "Microsoft.Batch/batchAccounts",
+                                                     "SHARED_RESOURCE"
+      )
+      relayNamespace <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                   "Microsoft.Relay/namespaces",
+                                                   "SHARED_RESOURCE"
+      )
+      storageAccountName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                       "Microsoft.Storage/storageAccounts",
+                                                       "SHARED_RESOURCE"
+      )
+      vnetName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                             "Microsoft.Network/virtualNetworks",
+                                             "SHARED_RESOURCE"
+      )
+      batchNodesSubnetName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                         "Microsoft.Network/virtualNetworks/subnets",
+                                                         "WORKSPACE_BATCH_SUBNET"
+      )
+      aksSubnetName <- getLandingZoneResourceName(lzResourcesByPurpose,
+                                                  "Microsoft.Network/virtualNetworks/subnets",
+                                                  "AKS_NODE_POOL_SUBNET"
+      )
+    } yield LandingZoneResources(
+      AKSClusterName(aksClusterName),
+      BatchAccountName(batchAccountName),
+      RelayNamespace(relayNamespace),
+      StorageAccountName(storageAccountName),
+      NetworkName(vnetName),
+      SubnetworkName(batchNodesSubnetName),
+      SubnetworkName(aksSubnetName)
+    )
+
+  private def getLandingZoneResourceName(landingZoneResourcesByPurpose: List[LandingZoneResourcesByPurpose],
+                                         resourceType: String,
+                                         purpose: String
+  ): F[String] = {
+    val lzResourcesOpt =
+      landingZoneResourcesByPurpose.find(lzResourceByPurpose => lzResourceByPurpose.purpose.equals(purpose))
+    val resourceOpt = lzResourcesOpt
+      .flatMap(lzResources =>
+        lzResources.deployedResources.find(lzResource => lzResource.resourceType.equals(resourceType))
+      )
+    val resourceNameOpt = resourceOpt.map(lzResource => lzResource.resourceName)
```

Comment:
See my sample API response above. Not all resource types have a `resourceName`. Perhaps we want this method to return `F[LandingZoneResource]` and have dedicated handling for each.